### DISCUSSION TITLE
Human Factors research

### DISCUSSION THREAD

**Discussion by jdekarske** [nasa/openmct#4266] [Community Q&A] [answered]
Hi!
I'm looking to do some research that investigates how remote operators use interfaces such as OpenMCT. Before getting too far into the depths of the code base,

1. Are there ways to log interactions? (e.g. switching between layouts, other mouse clicks)
2. Can I lock creating new layouts?

Thanks for any pointers!

**Comment by jvigliotta** [comment]
Hi @jdekarske as far as locking new display layouts, I'm not aware of a way to do that in the UI, but you can can change this line to false: https://github.com/nasa/openmct/blob/38db8f7fe5d166453c67fa83445c98737cdc0d00/src/plugins/displayLayout/DisplayLayoutType.js#L27 it won't show up in the create menu.

As for logging, I'm don't believe we have built in logging, but I will at @akhenry to see if he has a definitive answer.

**Comment by akhenry** [comment]
We do not do any logging of user interactions. That sort of instrumentation is available from specialist tools which should be relatively straightforward to incorporate into a build of Open MCT. @unlikelyzero may have some more specific guidance here.

**Comment by unlikelyzero** [comment, marked as answer]
@jdekarske there are a lot of great options for User Session Monitoring out there. Internally, we've done PoCs with Fullstory and LogRocket. Unfortunately, there don't seem to be many good options for self-hosting or even open source alternatives.

I would check the licensing and integration options for the different User Session Monitoring tools to find one that best suites your needs. As far as I know, every option will have a free trial with support to get you online within a day!
